### PR TITLE
Use proper package for @MongoClientName

### PR DESCRIPTION
**Pull request by geoand**
For now we just deprecate the old one annotation and support both

Relates to: https://github.com/quarkusio/quarkus/pull/11568#discussion_r478327219

### CODE REVIEW

File: docs/src/main/asciidoc/mongodb.adoc [hunk 1]
```diff
@@ -275,7 +275,11 @@ MongoClient mongoClient1;
 @Inject
 @MongoClientName("inventory")
 ReactiveMongoClient mongoClient2;
---
+----
+
+[WARNING]
+====
+The old `io.quarkus.mongodb.runtime.MongoClientName` has been deprecated. It still works, but will soon be removed, therefore users are encouraged to not use it.
```
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb.adoc:282]: You're missing the end of the WARNING.

File: extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/MongoClientName.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Marker annotation to select mongo connection of cluster configuration
+ * Use name parameter to select it
```
Comment (gsmet) [inline extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/MongoClientName.java:13]: I think we should improve this Javadoc.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added 2 small comments.

File: docs/src/main/asciidoc/mongodb.adoc (hunk 1 above)
Comment (geoand) [reply docs/src/main/asciidoc/mongodb.adoc:282]: Fixed and changed to `NOTE` as that is what is used in an already existing deprecation message in the doc

File: extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/MongoClientName.java (hunk 2 above)
Comment (geoand) [reply extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/MongoClientName.java:13]: Fixed

**Comment by geoand** [comment]
Seems like one of the tests failed, I need to fix it

**Comment by gsmet** [comment]
Rebased and force pushed to get the CI fix.

**Review by gsmet — APPROVED** [review]

**Pull request merged**